packages/core/src/governance/client/hooks/useAgreementMutations.web3.rsc.ts

```diff
@@ -1,33 +1,42 @@
 'use client';
 
 import useSWRMutation from 'swr/mutation';
-import { Config, writeContract } from '@wagmi/core';
+import useSWR from 'swr';
+import { useSmartWallets } from '@privy-io/react-auth/smart-wallets';
+import { encodeFunctionData } from 'viem';
+
 import {
   createProposal,
   getProposalFromLogs,
   mapToCreateProposalWeb3Input,
 } from '../web3';
+
 import { schemaCreateProposalWeb3 } from '@core/governance/validation';
-import useSWR from 'swr';
 import { publicClient } from '@core/common/web3/public-client';
-import { encodeFunctionData } from 'viem';
+
 import {
   agreementsImplementationAbi,
   agreementsImplementationAddress,
   daoProposalsImplementationAbi,
   daoProposalsImplementationAddress,
 } from '@core/generated';
 
-export const useAgreementMutationsWeb3Rpc = (config?: Config) => {
+export const useAgreementMutationsWeb3Rpc = () => {
+  const { client } = useSmartWallets();
+
   const {
     trigger: createAgreementMutation,
     reset: resetCreateAgreementMutation,
     isMutating: isCreatingAgreement,
     data: createAgreementHash,
     error: errorCreateAgreement,
   } = useSWRMutation(
-    config ? [config, 'createProposal'] : null,
-    async ([config], { arg }: { arg: { spaceId: number } }) => {
+    client ? ['smart-wallet', 'createProposal'] : null,
```

Comment:
This isn't a good cache key. You don't use the parameter anyway and it should be specific to the request, meaning including some parameter that make this request unique.